src/main/java/org/jabref/logic/ai/chathistory/BibDatabaseChatHistoryManager.java

```diff
@@ -0,0 +1,143 @@
+package org.jabref.logic.ai.chathistory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.desktop.JabRefDesktop;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageType;
+import dev.langchain4j.data.message.UserMessage;
+import jakarta.annotation.Nullable;
+import org.h2.mvstore.MVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class stores the chat history with AI for .bib files. The chat history file is stored in a user local folder.
+ * <p>
+ * It uses MVStore for serializing the messages. In case any error occurs while opening the MVStore,
+ * the class will notify the user of this error and continue with in-memory store (meaning all messages will
+ * be thrown away on exit).
+ *
+ * @implNote If something is changed in the data model, increase {@link org.jabref.logic.ai.AiService#VERSION}
+ */
+public class BibDatabaseChatHistoryManager implements AutoCloseable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BibDatabaseChatHistoryManager.class);
+    private static final String CHAT_HISTORY_FILE_NAME = "chatHistory.mv";
+
+    private final MVStore mvStore;
+
+    private final Map<Integer, String> messageLibrary;
+    private final Map<Integer, String> messageCitationKey;
+    private final Map<Integer, String> messageType;
+    private final Map<Integer, String> messageContent;
+
+    private final Map<Path, BibDatabaseChatHistory> bibDatabaseChatHistoryMap = new HashMap<>();
+
+    public BibDatabaseChatHistoryManager(DialogService dialogService) {
+        @Nullable Path ingestedFilesTrackerPath = JabRefDesktop.getAiFilesDirectory().resolve(CHAT_HISTORY_FILE_NAME);
+
+        try {
+            Files.createDirectories(JabRefDesktop.getAiFilesDirectory());
+        } catch (IOException e) {
+            dialogService.showErrorDialogAndWait("An error occurred while creating directories for storing chat history. Will store history in RAM", e);
+            ingestedFilesTrackerPath = null;
+        }
+
+        MVStore mvStore;
+
+        try {
+            mvStore = MVStore.open(ingestedFilesTrackerPath == null ? null : ingestedFilesTrackerPath.toString());
+        } catch (Exception e) {
+            dialogService.showErrorDialogAndWait("An error occurred while creating file for storing chat history. Will store history in RAM", e);
```

Comment:
UX pattern: If the user has some choice: Open a dialog. - If the user has no choice: Just notify them. In this case, just an error is shown and then the app is continuing. Why not continuing directly?

```java
            LOGGER.error("An error occurred while creating directories for storing chat history. Will store history in RAM", e);
            dialogService.notify("An error occurred while creating directories for storing chat history. Will store history in RAM");
```

Reply:
No tech details. Better: "An error occurred while creating directories for storing chat history. Chat history won't be remembered in next session" or something alike.